### PR TITLE
Fix #3057: Create a BUILD.bazel file for the persistence package in the data module

### LINKED ISSUE #3057: Create a BUILD.bazel file for the persistence package in the data module
This task is part of the build system migration from Gradle to Bazel. 

## Prerequesites

- Make sure you are able to build Oppia using Bazel
- Follow the [Oppia Bazel Setup instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) to setup Bazel and learn the terminology.
- Read the [Gradle Bazel Migration Best Practies and FAQ](https://github.com/oppia/oppia-android/wiki/Gradle--Bazel-Migration-Best-Practices-and-FAQ) for more context and see if there's an answer there to any question you have.

## Task

1. Create a BUILD.bazel file in

`data/src/main/java/org/oppia/android/data/persistence`

2. Move the library declaration from the root BUILD file to the new BUILD file

```python
# data/BUILD.bazel
load("@dagger//:workspace_defs.bzl", "dagger_rules")
# Remove this import in data/BUILD.bazel
load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")

# Remove this section from the data/BUILD.bazel
kt_android_library(
    name = "persistent_cache_store",
    srcs = ["src/main/java/org/oppia/android/data/persistence/PersistentCacheStore.kt"], # Update this line.
    custom_package = "org.oppia.android.data.persistence",
    manifest = "src/main/AndroidManifest.xml",
    visibility = ["//visibility:public"], # Update visibility.
    deps = [
        ":dagger",
        "//model:profile_java_proto_lite",
        "//third_party:androidx_appcompat_appcompat",
        "//third_party:androidx_lifecycle_lifecycle-livedata-ktx",
        "//third_party:com_caverock_androidsvg-aar",
        "//third_party:com_github_bumptech_glide_glide",
        "//utility",
        "//utility/src/main/java/org/oppia/android/util/profile:directory_management_util",
    ],
)

dagger_rules()
```

3. Add the library to the list of deps dependencies in the targets that require the library

4. Verify that the app builds

## Useful links

- [BUILD.bazel files](https://github.com/oppia/oppia-android/search?q=filename%3ABUILD.bazel&type=code)
- [Visibilities](https://github.com/oppia/oppia-android/blob/76dc13dba3ad77aecd2218a5372949d6551f12fe/BUILD.bazel)
- [third_party dependencies](https://github.com/oppia/oppia-android/blob/a70d9b016f0d77dd30c4939041f82ea02ee28d1d/third_party/versions.bzl)

### PR DESCRIPTION
**Pull request by jonathanalvares9009**
Created a BUILD.bazel file for the persistence package in the data module and added it to its dependents.
Fixes #3057 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by fsharpasharp** [comment]
Looks good, but you will have to replace all the old references to //data:persistent_cache_store.

I'm also thinking the new name of the library can be cache_store, since it's in the persistence directory.

**Comment by jonathanalvares9009** [comment]
@fsharpasharp name it cache_store

**Review by BenHenning — APPROVED** [review]
This is a really nice reorganization of the utility--thanks!

**Review by BenHenning — CHANGES_REQUESTED** [review]
@jonathanalvares9009 there seem to be lint failures: https://github.com/oppia/oppia-android/runs/2305408801?check_suite_focus=true. Please address these & assign back.

File: data/BUILD.bazel [hunk 1]
```diff
@@ -4,31 +4,12 @@ This library provides data to the rest of the application.
...
-        "//utility",
-        "//utility/src/main/java/org/oppia/android/util/profile:directory_management_util",
-    ],
-)
-
 # keep sorted
 TEST_DEPS = [
     ":dagger",
-    ":persistent_cache_store",
+    "//data/src/main/java/org/oppia/android/data/persistence:cache_store",
```
Comment (fsharpasharp) [inline data/BUILD.bazel]: Put this line after the other two to keep it sorted -- it will fix the linting issue.

Comment (jonathanalvares9009) [reply data/BUILD.bazel]: Done. Thanks will remember this.

**Review by BenHenning — APPROVED** [review]
Thanks @jonathanalvares9009! This LGTM.

**Pull request merged**